### PR TITLE
fix: validate PGBOUNCER_URLS on config gen

### LINKED ISSUE #171: Buildpack should validate `PGBOUNCER_URLS` explicitly
Currently if `PGBOUNCER_URLS` is set on an app, and some of the env vars it refers to don't exist, then empty entries are added by `gen-pgbouncer-conf.sh` to `/app/vendor/pgbouncer/pgbouncer.ini`, like so:

```
[databases]
db1= host=REDACTED.compute-1.amazonaws.com dbname=REDACTED port=5432
db2= host= dbname= port=
db3= host=REDACTED.compute-1.amazonaws.com dbname=REDACTED port=5432
```

As of newer pgbouncer versions (such as v1.17.0, which #170 recently updated Heroku-18/20 to), this then results in a confusing error message when pgbouncer tries to read the config:

```
ERROR syntax error in connection string
ERROR invalid value "host= dbname= port=" for parameter db2 in configuration
FATAL cannot load config file
```

In this case the "db2" refers to the second database listed in `PGBOUNCER_URLS`.

So if `PGBOUNCER_URLS` was set to `DATABASE_FOO_URL DATABASE_BAR_URL DATABASE_BAZ_URL`, the above error means the `DATABASE_BAR_URL` env var is not set, or was set to the empty string.

The fix is for the `PGBOUNCER_URLS` env var to be updated on the app so it doesn't refer to non-existent DB env vars.

To improve the UX here, `gen-pgbouncer-conf.sh` should really check for empty values here, and fail early with a clear error message:
https://github.com/heroku/heroku-buildpack-pgbouncer/blob/92598696b88b522dd792861fdf795372b9f3a537/bin/gen-pgbouncer-conf.sh#L47-L49

### PR DESCRIPTION
**Pull request by mble-sfdc**
Invalid config can sneak in, resulting in confusing error messages. This commit attempts to validate entries in `PGBOUNCER_URLS` before passing them to the config. It handles two cases - the empty case, and the invalid PostgreSQL connection string case.

Fixes #171

### CODE REVIEW

**Comment by edmorley** [comment]
@mble-sfdc I'm not planning on working on this (I don't have the cycles), so I'm puzzled by the assignment? Did you mean to request review instead (which I'd be happy to do; albeit it won't be too in depth)? :-)

**Comment by mble-sfdc** [comment]
@edmorley Ah, yes, I meant review. Apologies.

File: README.md
Comment (edmorley) [inline README.md:114]:
```suggestion
> ⚠️ A referenced configuration variable in `PGBOUNCER_URLS` must not be empty, and must be a valid PostgreSQL connection string.
```

**Review by edmorley — APPROVED** [review]
Looks good at a quick glance - but wasn't a thorough review.

Has it been tested with a real Heroku app for both success and failing cases?

**Comment by mble-sfdc** [comment]
@edmorley not yet - happy to block until I can test.

**Comment by edmorley** [comment]
Happy to leave it to your judgement :-) 

After the merge of this PR, a new version of the buildpack will need to be released to the buildpack registry. @beanieboi has access to do this, otherwise it's mostly the languages team that have access - to ping one of us and we can publish for you.

**Comment by mble-sfdc** [comment]
@edmorley Finally had the opportunity to test this one - all good.

**Comment by edmorley** [comment]
Feel free to merge whenever you like :-)

**Comment by mble-sfdc** [comment]
@edmorley I actually can't - protected branch and I'm not authorised :D

**Comment by edmorley** [comment]
I've switched the repo to the new data team name - should work now :-)

**Pull request merged**
